src/codeu/chat/client/simplegui/ChatGuiFX.java

```diff
@@ -319,10 +325,16 @@ private void addConversation(ActionEvent e) {
             String name = dialog.showAndWait().get();
 
             // TODO: check for duplicate conversations & handle them
+            for (final ConversationSummary cs : clientContext.conversation.getConversationSummaries()) {
```

Comment:
Is this the most efficient way to check whether the title is already used? Is there a Map or something you could be using instead? What if there were a million conversations?